### ISSUE TITLE
DGCNN实现疑惑

### ISSUE THREAD

**Issue by snailpt** [XJTU-EEG/LibEER#17]
衷心地感谢你们的开源工作。
https://github.com/XJTU-EEG/LibEER/blob/main/LibEER/models/DGCNN.py里关于切比雪夫多项式的实现细节有点疑惑，希望帮忙解答下。
问题1：
T0(x)=1，T1(L)=L, 所以当k=2时，chebyshev_polynomial多项式应该用torch.cat((x.unsqueeze(1), torch.matmul(lap, x).unsqueeze(1)), dim=1)才是和论文匹配的？

    def chebyshev_polynomial(self, x, lap):
        """
        calculate the chebyshev polynomial
        :param x : input x
        :param lap: the input laplacian matrix
        :return: the chebyshev polynomial components
        """
        t = torch.ones(x.shape[0], x.shape[1], x.shape[2]).to(x.device)
        if self.k == 1:
            return t.unsqueeze(1)
        if self.k == 2:
            return torch.cat((t.unsqueeze(1), torch.matmul(lap, x).unsqueeze(1)), dim=1)

问题2： 目前的代码实现用的是归一化拉普拉斯，而论文里用的是L*~=2L*/landa(max)* - I, 下面的函数如果修改下是否会更贴近论文的实现：
def laplacian(w):
    """
    calculate the laplacian of the adjacency matrix
    :param w: the adjacency matrix
    :return: l: the normalized Laplacian matrix
    """
    # d is the sum of each row of a matrix.
    d = torch.sum(w, dim=1)
    # reciprocal square root of a vector
    d_re = 1 / torch.sqrt(d + 1e-5)
    # create a matrix with the d_re vector as its diagonal elements
    d_matrix = torch.diag_embed(d_re)
    # calculate the laplacian matrix
    lap = torch.eye(d_matrix.shape[0], device=w.device) - torch.matmul(torch.matmul(d_matrix, w), d_matrix)
    return lap
如果在return 前加上这几行代码是否更贴近论文里的实现？
I = torch.eye(w.size(-1), device=w.device, dtype=w.dtype)
lambda_max = torch.linalg.eigvals(lap).real.max()
L_tilde = (2.0 / lambda_max) * lap - I
return L_tilde

**Comment by ButterSen** [comment]
问题1貌似确实是这样，复现好像在这里有点问题，谢谢您指出问题，我会更新此问题！

**Comment by snailpt** [comment]
非常感谢您的回复！对于问题2，DGCNN里原文是这么说的：<img width="543" height="519" alt="Image" src="https://github.com/user-attachments/assets/4e17dba1-462f-47d6-ae58-6f8034801fc7" />
感觉目前没有做rescaled Laplacian，按下面这个实现看上去应该更贴近论文，只是不清楚工程上直接用lambda_max =2.0是不是更容易训练还是用算出来的lamba取最大效果好，希望给点建议。
lap = laplacian(adj)                 # L
lambda_max = 2.0                     # 或者实际估计最大特征值
lap_tilde = 2 * lap / lambda_max - torch.eye(lap.shape[0], device=lap.device)
